activerecord/test/cases/encryption/configurable_test.rb

```diff
@@ -55,7 +55,7 @@ class ActiveRecord::Encryption::ConfigurableTest < ActiveRecord::EncryptionTestC
   end
 
   test "installing autofiltered parameters will work with unnamed classes" do
-    application = OpenStruct.new(config: OpenStruct.new(filter_parameters: []))
+    application = ActiveSupport::OrderedOptions[config: ActiveSupport::OrderedOptions[filter_parameters: []]]
```

Comment:
I'm not too sure how I feel about using OrderedOption as a stand-in for OpenStruct.

From a purely pragmatic point of view, it's definitely superior as at least it's a class we control and we know it's loaded.

But I feel in many cases here, and espeically on the code just above, it may make more sense to use a `Minitest::Mock` as it is closer to the intent. WDYT?

Reply:
Turned out, that `OpenStruct` is not equivalent to `ActiveSupport::OrderedOptions`, as failing CI shows. The problem is described in https://github.com/rails/rails/issues/51469. `ActiveSupport::OrderedOptions#respond_to_missing?` returns `true` for all the calls and so the line https://github.com/rails/rails/blob/bad7ff1664fb05cc227d0386ee3cbe4c292efe05/activerecord/lib/active_record/scoping/default.rb#L161 is not working correctly.

Just tried to convert to use mocks, and turned out that methods like `is_a?` should be mocked too. Meh. So probably just `require "ostruct"` in all the places?

Reply:
> So probably just require "ostruct" in all the places?

Meh, I think the issue will just crop back up.

I'm tempted to suggest using `Struct`, it's a tiny bit more verbose but would do the job fine.

Reply:
@byroot  Updated to use `Struct`s. Please take a look.